pkg/kube/cluster-init.sh

```diff
@@ -614,6 +666,17 @@ check_cluster_config_change() {
 
             logmsg "provision config file for node to cluster mode"
             provision_cluster_config_file true
+            provision_status=$?
+
+            # If in the middle of waiting for bootstrap node to be ready, the node is converted again to single node
+            # we need to get out of this loop and go back to single node mode
+            if [ $provision_status -eq 1 ]; then
+              logmsg "EdgeNodeClusterStatus file disappeared, reset the status and back to single node and reboot"
+              rm /var/lib/base-k3s-mode
+              touch /var/lib/convert-to-single-node
+              reboot
```

Comment:
What is the RebootReason published by edge node when reboot is triggered by cluster-init.sh?